src/canvas/renderCanvas.js

```diff
@@ -76,18 +79,126 @@ import {REACT_TOTAL_NUM_LANES} from '../constants';
 //                  '──────────────────────────
 //
 
+// TODO: (windowing, optimization) We can avoid rendering offscreen data in many
+// of the render* functions in this file.
+
 function renderBackgroundFills(context, canvasWidth, canvasHeight) {
   // Fill the canvas with the background color
   context.fillStyle = COLORS.BACKGROUND;
   context.fillRect(0, 0, canvasWidth, canvasHeight);
 }
 
-/**
- * Render all charting data (once it's loaded and processed) within the
- * "scrollable" region.
- */
-// TODO (windowing) We can avoid rendering all of this if we've scrolled some of it off screen.
-function renderReactMarksAndMeasures(
+function renderReactEventRow(
```

Comment:
Heading comments describing the purpose of the function would be nice here

Reply:
Added!